app/controllers/flowchart_node_controller.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+class FlowchartNodeController < ApplicationController
+  def create
+    prev_id = params[:prev_id]
+    is_child = params[:is_child]
+    begin
+      prev_node = FlowchartNode.find(prev_id)
+    rescue ActiveRecord::RecordNotFound
+      render status: 404, json: { error: "No node found with id #{prev_id}." }
```

Comment:
Why not let the controller return a 404 on the error rather than rescuing the error?

Reply:
I do this so we can control the error message and status that was being returned. I can remove it if just returning 404 on the error is better.